File: backend/models/entities/community.js
import { Model } from 'sequelize'

export default (sequelize, DataTypes) => {
  class Community extends Model {
    static associate(models) {
      // Community belongs to Category
      Community.belongsTo(models.Category, {
        foreignKey: 'categoryId',
        as: 'category',
      })

      // Community has many Chats
      Community.hasMany(models.Chat, {
        foreignKey: 'communityId',
        as: 'chats',
      })

      // Community has many Users (many-to-many)
      Community.belongsToMany(models.User, {
        through: models.JoinCommunity,
        foreignKey: 'communityId',
        otherKey: 'userId',
        as: 'members',
      })
    }
  }

  Community.init(
    {
      communityId: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
        allowNull: false,
      },
      communityName: DataTypes.STRING,
      categoryId: DataTypes.INTEGER,
      avatarId: DataTypes.STRING,
      memberNum: DataTypes.INTEGER,
      description: DataTypes.STRING(1500),
    },
    {
      sequelize,
      modelName: 'Community',
      tableName: 'community',
      timestamps: false,
    }
  )

  return Community
}

File: frontend/app/src/main/java/com/example/frontend/data/model/Community.kt
package com.example.frontend.data.model

import com.google.gson.annotations.SerializedName

data class Community(
    @SerializedName("communityId") val id: Int,
    @SerializedName("communityName") val name: String,
    @SerializedName("category") val category: Category,
    var avatarUrl: String? = null,
    @SerializedName("memberNum") val memberNum: Int,
    val description: String? = null,
    val members: List<User>? = null
){
    public fun copy():Community{
        return Community(
            id = this.id,
            name =  this.name,
            category =  this.category,
            avatarUrl =  this.avatarUrl,
            memberNum =  this.memberNum,
            members =  this.members,
        )
    }
}